proxy.go

```diff
@@ -747,7 +618,7 @@ func loadSiteCredentialsFromEnvVar(envVar string) []string {
 
 // New is the Proxy factory. Errors can be introspected, and provide contextual
 // information.
-func New(localProxyURI, upstreamProxyURI, pacURI string, pacProxiesCredentials []string, options *Options) (*Proxy, error) { //nolint // FIXME Function 'New' has too many statements (67 > 40) (funlen); calculated cyclomatic complexity for function New is 24, max is 10 (cyclop)
+func New(localProxyURI, upstreamProxyURI, pacURI string, pacProxiesCredentials []string, options *Options, log Logger) (*Proxy, error) { //nolint // FIXME Function 'New' has too many statements (67 > 40) (funlen); calculated cyclomatic complexity for function New is 24, max is 10 (cyclop)
```

Comment:
this changes the external API. The `New` function is used to initialize Forwarder in its parent app. AFAICT, before, the logger parameter was optional (and it had to be `sypl.Sypl`). With this change, the logger is generic but it also implies changes elsewhere

Reply:
Whatever uses the package uses the `forwarder/proxy::New` and would need to be migrated anyways.
This is not a big deal since they are most likely bound to a commit, as long as we do not mess up with git history we are good.

We can also introduce [v2](https://therootcompany.com/blog/bump-go-package-to-v2/) it's messy but can be done.